core-js/src/main/javascript/cdf-legacy/queries/coreQueries.js

```diff
@@ -564,6 +564,12 @@
         queryDefinition.settingcolumnHeaders = options.columnHeaders;
       }
 
+      // only relevant for a component with server side pagination
+      if(options.exportPage !== true) { // only export a page explicitly
```

Comment:
in order to preserve the default in all cases, change this condition to `=== false`
The comment can be updated accordingly `// only export the entire dataset when exportPage is false`